app/models/merchant.rb

```diff
@@ -40,7 +40,22 @@ def filter_item_status(status_enum)
     items.where(status: status_enum)
   end
 
+  def self.top_five_merchants
+    joins(invoices: :invoice_items)
+      .joins(:transactions)
+      .where(transactions: { result: 1 })
+      .select("merchants.*, SUM(invoice_items.unit_price * invoice_items.quantity) AS merchant_revenue")
+      .group(:id)
+      .limit(5)
+      .order(merchant_revenue: :desc)
+  end
+
+
   def self.filter_merchant_status(status_enum)
     where(status: status_enum)
   end
+
+  def revenue_by_merchant
+    Merchant.joins(:invoice_items).where(id: self.id).sum("invoice_items.unit_price * invoice_items.quantity")
```

Comment:
Seeing this as Merchant instead of merchants reminded me how calling the model like Merchant is an active record thing, where as calling the table as merchants is a SQL thing.  I have to go back and check my AR methods to make sure they are calling the model instead of the table.